plugin-cache/index.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var EventEmitter = require('events').EventEmitter;
+var Q = require('q');
+var util = require('util');
+
+var getList = require('./src/get');
+var updateList = require('./src/update');
+
+var Logger = process.env.Logger || {
+  log: console.log.bind(console),
+  error: console.error.bind(console)
+};
+
+var Q = process.env.Q || Q;
+
+var Cache = function () {
+  this.list = [];
+
+  this.q = Q;
+
+  this.get = getList.bind(this);
```

Comment:
Well, I think event emitters can be useful, but they usually end up creating memory leaks all over the place. Ideally I'd stay away from that pattern inside a server component.